### ISSUE TITLE
Configuration File is not Saving/Loading Properly

### ISSUE THREAD

**Issue by mattsilver1** [Dragory/modmailbot#98]
```{
  "token": "",
  "mailGuildId": "464234238443520000",
  "mainGuildId": "422805784334958603",
  "logChannelId": "464235042151858176",

  "status": "Message me for help!",
  "responseMessage": "Thank you for your message! Our team will reply to you here as soon as possible.",

 "alwaysReplyAnon": "true",
 "typingProxy": "true",
 "useNicknames": "true",
 "allowMove": "true",
 "enableGreeting": "true",
 "greetingMessage": "Welcome to Digital Marketing - White Hat & Black Hat, if you ever need any help contacting the staff members, please send me a DM!",
 "newThreadCategoryId": "464235011248488458",
 "mentionRole": "<@&467566988495945739>",
 "url": "server.corporate.tssservicecollc.com",
 "closeMessage": "Your thread has been closed by the system, if you need anything else please message me again.",
 "relaySmallAttachmentsAsAttachments": "true",
 "threadTimestamps": "true",

}
```
Having an issue with this, this is the config we want to work but for some reason it's not correctly loading the new settings "mentionRole and down" I don't understand why it's not "saving" them but, anything you can do to help?

**Comment by judge2020** [comment]
See the function getInboxMention:

https://github.com/Dragory/modmailbot/blob/999bccd7aabec7a970bbf6dfbb4eeb8e927c276e/src/utils.js#L239-L244

The bot already inserts the `<&>`. so the value of mentionRole should only be the role ID (467566988495945739).

**Comment by mattsilver1** [comment]
Yeah I just changed it to that and the bot isn't saving the new settings.

**Comment by Dragory** [comment]
The bot doesn't handle saving new settings. It loads the config file when it starts, so if you change settings, you need to restart the bot for them to apply.

**Comment by mattsilver1** [comment]
I've terminated the batch job (restarted) it like around 10 times probably and it's not keeping my changes.

**Comment by mattsilver1** [comment]
So I did a little bit of digging, and the bot is running on Modmail v2.11.1
I'm actually assuming this is an issue since I checked your release information, would we just update the bot with ``npm install`` or how would that be done?

**Comment by Dragory** [comment]
By not "keeping" your changes, you mean it's not loading them? Or is it reverting the config file?

**Comment by mattsilver1** [comment]
It's running 2.11.1 & It's not loading them. It's not erasing the configuration file but the changes I've implemented aren't exactly "loading/saving" if you understand what I mean.

**Comment by mattsilver1** [comment]
Alright, I reinstalled `npm install` and it's working as of now. But I noticed that it's not taking my argument for "closeMessage" it throws an error. I had to remove that line of code from the config file but is there a particular reason why it's throwing an error for "closeMessage"? It basically says it's an invalid option.

**Comment by JonasPardon** [comment]
@mattsilver1 the `closeMessage` is not there yet in the 2.11.1 release, you'll have to clone the master branch for that to work.

**Comment by mattsilver1** [comment]
Hmm, I just re-cloned the master branch again and it still appears not to be operational.

**Comment by JonasPardon** [comment]
Try just doing `git clone https://github.com/Dragory/modmailbot`, not sure on how to overwrite an existing folder. You could just make a copy of your old folder as a backup, cloning the master branch with said command and filling in your config file. It might also fix your other problem if you start off fresh.

**Comment by JonasPardon** [comment]
Then whenever there is an update to the master branch, you can just do `git pull` to update it

**Comment by mattsilver1** [comment]
Is there a way to move the data "logs and stuff" over to the new clone I just made on a separate file? And if so, what files would I be moving?

**Comment by JonasPardon** [comment]
I'm not sure and I'm on phone away from home right now, but try copying the 'db' and 'logs' folders that are in the root folder.

**Comment by mattsilver1** [comment]
Alright, thanks for your fast responses and help. I'm currently restarting the server to shut down the bot (running the bot as a hidden system service) so I can't exactly stop it from running. I'll let you know what occurs when I copy the DB files.
